### PR TITLE
Fix ContentLengthReadAsyncSingleBytesAtATime

### PR DESCRIPTION
**Pull request by halter73**
Addresses #23628

### CODE REVIEW

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs [hunk 1]
```diff
@@ -1048,27 +1048,52 @@ public async Task ContentLengthReadAsyncPipeReaderBufferRequestBodyMultipleTimes
         }
 
         [Fact]
-        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/23628")]
```
Comment (Tratcher) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:1051]: Did this repro 100% or was it flaky. If it was flaky then keep it Quarantined for 30 days.

Comment (halter73) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:1051]: I think this is a clear enough fix that we can remove the QuarantinedTest attribute immediately, but I can leave it quarantined and let it become unquarantined naturally.

Comment (halter73) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:1051]: I made another comment before seeing this. I think the issue is pretty obviously fixed, but I can add the attribute back.

> I think this is a clear enough fix that we can remove the QuarantinedTest attribute immediately, but I can leave it quarantined and let it become unquarantined naturally.

Comment (BrennanConroy) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:1051]: I'm happy either way if we're confident.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**